qemu/tests/qemu_guest_agent.py

```diff
@@ -429,12 +434,46 @@ def gagent_check_sync(self, test, params, env):
 
         :param test: kvm test object
         :param params: Dictionary with the test parameters
-        :param env: Dictionary with test environmen.
+        :param env: Dictionary with test environment.
         """
         error_context.context("Check guest agent command 'guest-sync'",
                               logging.info)
         self.gagent.sync()
 
+    @error_context.context_aware
+    def gagent_check_guest_info(self, test, params, env):
+        """
+        Execute "guest-info" command to guest agent
+
+        Test steps:
+        1) Send "guest-info" command and check the version of qga.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Check guest agent command 'guest-info'",
+                              logging.info)
+        qga_ver_qga = self.gagent.guest_info()['version'].strip()
+        if params.get("os_type") == 'windows':
+            win_qgabuild_guest = session.cmd_output(params["cmd_qga_build"]
+                                                    ).strip()
+            if qga_ver_qga != win_qgabuild_guest:
+                test.fail("The qga build %s from qga is different with %s"
+                          " from windows guest." % (qga_ver_qga,
+                                                    win_qgabuild_guest))
+        else:
+            qga_ver_guest_raw = str(self.qga_v)
+            pattern = r"(\d+.\d+.\d+)"
+            qga_ver_guest = re.findall(pattern, qga_ver_guest_raw, re.I)[0]
+            if qga_ver_qga != qga_ver_guest:
+                test.fail("The qga version %s from qga is different with %s "
+                          "from linux guest." % (qga_ver_qga, qga_ver_guest))
```

Comment:
```suggestion
        if params.get("os_type") == 'windows':
            qga_ver_guest = session.cmd_output(params["cmd_qga_build"]
                                                    ).strip()
        else:
            qga_ver_guest_raw = str(self.qga_v)
            pattern = r"(\d+.\d+.\d+)"
            qga_ver_guest = re.findall(pattern, qga_ver_guest_raw, re.I)[0]
        if qga_ver_qga != qga_ver_guest:
            test.fail("The qga version %s from qga is different with %s "
                      "from guest." % (qga_ver_qga, qga_ver_guest))
```

Reply:
The original thought was to distinguish between build and version. but this is better for code clean. thanks